### DISCUSSION TITLE
[Question]: Setting Max Tokens

### DISCUSSION THREAD

**Discussion by rrdd** [danny-avila/LibreChat#3802] [Q&A]
### What is your question?

Where can i set the system wide max tokens for open ai for example or for a model?

### More Details

to protect expensive tasks i want to limitmax tokens per model oder provider.

### Code of Conduct

- [X] I agree to follow this project's Code of Conduct

**Comment by danny-avila** [comment]
You will need to make a hard-coded change to the codebase

`api/utils/tokens.js`

This will be something that is configurable via librechat.yaml soon enough

**Reply by bradydowling** [reply]
After I update the line item in `tokens.js` I'm actually still getting the 8187 token limit error. Maybe I'll just wait until this is configurable via `librechat.yaml` 🤷‍♂️ 

By the way, congratulations and thank you for much for your work on LibreChat. You should be really proud of what you've done with it, it's really incredible!

**Comment by rrdd** [comment]
Thank you for your fast reply.

I modified in the file api/utils/tokens.js to:
  'gpt-4o': 4000, // -10 from max


means max 4000 token in and out? can i check the value on backend side to make sure?

additional note to the settings independent from the tokens.js file:
when setting token of the model in the frontend (Max. Kontext-Token, Max. Antwort-Token 5000 ) i always get an error: "Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request."

removing the entries is not possible. it will not store empty value.

**Reply by danny-avila** [reply]
> means max 4000 token in and out? can i check the value on backend side to make sure?

No, that is for the context.

What you're looking for is here:

https://github.com/danny-avila/LibreChat/blob/f742b9972ec768e5b5076debc747d01d3fd77b2e/api/app/clients/OpenAIClient.js#L172

>  i always get an error: "Something went wrong. Here's the specific error message we encountered: An error occurred while processing your request."

Check the error from the backend logs, it will make sense

**Comment by rrdd** [comment]
my fault using 5000 in the frontend
This model supports at most 4096 completion toke... Sorry

**Comment by rrdd** [comment]
Last Info. It is not possible to delete the setting for max token in the frontend after you have set it.

**Reply by danny-avila** [reply]
Not sure what you mean, it is more than possible from my understanding. It's saved to the conversation after a successful message.

**Reply by rrdd** [reply]
i set max token context and max token for answer in the frontend under presettings to 2000. after that i can not remove this value anymore. the empty value is not stored. i have to set any number or it will not stored.
